### DISCUSSION TITLE
Cake.powershell - Issue upgrading builds from cake 0.38.5 to 1.10

### DISCUSSION THREAD

**Discussion by ChrisClark303** [cake-build/cake#3406] [Q&A]
Hi,

I wonder if anyone can help - we have a number of builds that have been using Cake 0.3x.x for some time and I'm trying to bring them up to the latest version. I have made slow progress on this but have hit a bit of brick wall with the cake.powershell addin. Every time I run the build it attempts to install this addin, downloads it successfully, but then gives me this error:

Could not find any assemblies compatible with .NETCoreApp,Version=v2.1.

I'm attempting to use cake.powershell 0.4.8:
#addin "nuget:?package=Cake.Powershell&version=0.4.8"

This hasn't changed since I upgraded the cake version (I have since tried a number of different version on cake.powershell, both higher and lower, and I get the same issue). From what I can see this version of the addin has no dependencies and should be suitable for core2.1 apps (and has indeed been working fine under Cake 0.38.5). 

We also use cake.sonar (1.1.25) and this still works perfectly. 

I'm running the builds via TFS 2017 with the Cake extension, but I get the same error if I run via command line.

Please let me know if you require any further information.

Thanks,

Chris

**Comment by pascalberger** [comment]
Looking at the [extension page of Cake.Powershell](https://cakebuild.net/extensions/cake-powershell/) you see supported Cake Versions and target frameworks for the latest version. Cake.Powershell 1.0.1 has been built against Cake 1.0, but requires netcoreapp3.1 or higher. Cake.Powershell < 1.0 (liket 0.4.8 you have used in the provided example) was neither built against Cake 1.x, nor does it have cross plattform support based on [release notes for Cake.Powershell 1.0](https://www.nuget.org/packages/Cake.Powershell/1.0.0).

While Cake .NET Tool and Cake Frosting [support .NET Core 2.1](https://cakebuild.net/docs/running-builds/runners/#supported-platforms), this specific addin does not.

That being said, the mentioned error can't be caused by just updating from Cake 0.3x to 1.0, but is more the result of switching the Cake runner. I assume you have used [Cake runner for .NET Framework 0.3x](https://cakebuild.net/docs/running-builds/runners/cake-runner-for-dotnet-framework) and updated to [Cake .NET Tool 1.x](https://cakebuild.net/docs/running-builds/runners/dotnet-tool). Your build using Cake.PowerShell wouldn't have worked either with Cake .NET Tool 0.3x, but should still work fine with  Cake runner for .NET Framework 1.x.
